storage/remote/queue_manager.go

```diff
@@ -887,14 +884,15 @@ func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.Ti
 		return err
 	}
 
-	span.SetTag("samples", len(samples))
-	span.SetTag("requestSize", len(req))
-
 	retries := 0
-	// Anonymous function is required so that retries value is not captured as zero.
-	defer func() { span.SetTag("retries", retries) }()
 
 	for {
+		span, ctx := opentracing.StartSpanFromContext(ctx, "Remote Send Batch")
+		span.SetTag("samples", len(samples))
+		span.SetTag("requestSize", len(req))
+		span.SetTag("retries", retries)
+		span.SetTag("remoteStoreName", s.qm.storeClient.Name())
```

Comment:
We use `remote_name` in the metrics, it would be nice to be consistent. Perhaps let's also do `snake_case` for all tags?

Reply:
we should maybe include remote url as well since name might not always be useful (user hasn't set a name in the config, so it defaults to the hash)?

Reply:
Good call. :-)